lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -228,8 +228,11 @@ static StringRef getPortVerilogName(Operation *module, size_t portArgNum) {
 /// Return the verilog name of the port for the module.
 static StringRef getInputPortVerilogName(Operation *module, size_t portArgNum) {
   auto hml = cast<HWModuleLike>(module);
-  return hml.getPort(hml.getHWModuleType().getPortIdForInputId(portArgNum))
-      .getVerilogName();
+  auto pId = hml.getHWModuleType().getPortIdForInputId(portArgNum);
+  if (auto attrs = dyn_cast_or_null<DictionaryAttr>(hml.getPortAttrs(pId)))
+    if (auto updatedName = attrs.get("hw.verilogName"))
+      return updatedName.cast<StringAttr>().getValue();
```

Comment:
This may be slightly more compact as:

```suggestion
    if (auto updatedName = attrs.getAs<StringAttr>("hw.verilogName"))
      return updatedName.getValue();
```

This trades a `cast` for a `dyn_cast_or_null` which may be slightly slower due to the extra non-assert checks.